massa-network/src/network_worker.rs

```diff
@@ -1045,6 +1027,83 @@ impl NetworkWorker {
         Ok(())
     }
 
+    /// Start to mock a handshake and send a message with a list of peer.
+    /// The function is used while `manage_in_connections()` if the current
+    /// node reach the number of maximum in connection.
+    ///
+    /// Timeout if the connection as not been done really quickly resolved
+    /// but doesn't throw the timeout error.
+    ///
+    /// Return an error ONLY if we get a `NetworkError`.
+    ///
+    /// ```bash
+    /// In connection node            Curr node
+    ///        |                          |
+    ///        |------------------------->|      : Try to connect but the
+    ///        |                          |        curr node reach the max
+    ///        |                          |        number of node.
+    ///        |                          |      : Connection success anyway
+    ///        |                          |        and the in connection enter
+    ///        |<------------------------>|        in `HandshakeWorker::run()`
+    ///        |  symetric read & write   |
+    ///```
+    ///
+    /// In the `symetric read & write` the current node simulate a handshake managed
+    /// by the *connection node* in `HandshakeWorker::run()`, the current node
+    /// send a ListPeer as a message.
+    async fn send_peer_list_in_handshake(
+        &self,
+        reader: ReadHalf,
+        writer: WriteHalf,
+        remote_addr: SocketAddr,
+    ) -> Result<(), NetworkError> {
```

Comment:
either:
* return an error in both cases (timeout AND read/write failure) but ignore it (or just print a debug message) in the caller (to not make the whole program crash !!!) 
* or do not return anything here and ignore errors (maybe with a debug message) within the send_peer_list_in_handshake function, and rename the function `try_send_peer_list_in_handshake` to make it clear that it might fail silently


I vote for solution 1

Reply:
I made a proposal